### PR TITLE
feat(VadeSecure): create parser

### PR DESCRIPTION
**Pull request by LeoArurault**
Creating VadeSecure M365 parser

### CODE REVIEW

**Review by squioc — DISMISSED** [review]
LGTM

**Comment by otetard** [comment]
test

File: VadeSecure/vade_secure_m365/tests/email.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "input": {
+        "sekoiaio": {
+            "intake": {
+                "dialect": "vade-m365",
+                "dialect_uuid": "e4a758fc-7620-49e6-b8ed-b7fb3d7fa232"
+            }
+        },
+        "message": " {\"id\": \"zekfnzejnf576rge8768\", \"date\": \"2022-02-10T13:00:05.454Z\", \"sender_ip\": \"192.168.1.1\", \"from\": \"test@sekoia.io\", \"from_header\": \"<test@sekoia.io>\", \"to\": \"test@vadesecure.com\", \"to_header\": \"\\\"test@vadesecure.com\\\" <test@vadesecure.com>\", \"subject\": \"Lorem ipsum dolor\", \"message_id\": \"<01de2305-f75b-49db-8c61-f661bd498e63.protection.outlook.com>\", \"urls\": [{\"url\": \"https://sekoia.io\"}], \"attachments\": [], \"status\": \"LEGIT\", \"substatus\": \"\", \"remediation_type\": \"none\", \"remediation_ids\": [], \"action\": \"NOTHING\", \"folder\": \"\", \"size\": 113475, \"current_events\": [], \"whitelisted\": false} "
+    },
+    "expected": {
```
Comment (PierrickV) [inline VadeSecure/vade_secure_m365/tests/email.json:11]:
```json
{
    "id": "zekfnzejnf576rge8768",
    "date": "2022-02-10T13:00:05.454Z",
    "sender_ip": "192.168.1.1",
    "from": "test@sekoia.io",
    "from_header": "<test@sekoia.io>",
    "to": "test@vadesecure.com",
    "to_header": "test@vadesecure.com <test@vadesecure.com>",
    "subject": "Lorem ipsum dolor",
    "message_id": "<01de2305-f75b-49db-8c61-f661bd498e63.protection.outlook.com>",
    "urls": [
      {
        "url": "https://sekoia.io"
      }
    ],
    "attachments": [],
    "status": "LEGIT",
    "substatus": "",
    "remediation_type": "none",
    "remediation_ids": [],
    "action": "NOTHING",
    "folder": "",
    "size": 113475,
    "current_events": [],
    "whitelisted": false
  }
  ```

I noted that some fields are not extracted (e.g. action, whitelisted,..) don't we need those fields ?

File: VadeSecure/vade_secure_m365/_meta/fields.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
+vadesecure.folder:
+    description: The name of the folder in the end user's mailbox if the message has been moved by Vade.
+    name: vadesecure.folder
+    short: vadesecure.folder
+    type: keyword
+vadesecure.from_header:
+    description: 'The email address declared in the From: header of the message, which may differ from the address used in the SMTP MAIL FROM command.'
```
Comment (PierrickV) [inline VadeSecure/vade_secure_m365/_meta/fields.yml:7]:
```suggestion
    description: The email address declared in the From: header of the message, which may differ from the address used in the SMTP MAIL FROM command.
```

File: VadeSecure/vade_secure_m365/_meta/fields.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    name: vadesecure.status
+    short: vadesecure.status
+    type: keyword
+vadesecure.substatus:
+    description: Filtering substatus
+    name: vadesecure.substatus
+    short: vadesecure.substatus
+    type: keyword
+vadesecure.to_header:
+    description: 'The email address declared in the To: header of the message, which may differ from the address used in the SMTP RCPT TO command.'
```
Comment (PierrickV) [inline VadeSecure/vade_secure_m365/_meta/fields.yml:27]:
```suggestion
    description: The email address declared in the To: header of the message, which may differ from the address used in the SMTP RCPT TO command.
```

File: VadeSecure/vade_secure_m365/_meta/fields.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    name: vadesecure.to_header
+    short: vadesecure.to_header
+    type: keyword
+
+vadesecure.reason:
```
Comment (PierrickV) [inline VadeSecure/vade_secure_m365/_meta/fields.yml:32]: event.reason ?  Shouldn't this be https://github.com/elastic/ecs/blob/8.0/generated/csv/fields.csv#L194
```suggestion
event.reason:
```

File: VadeSecure/vade_secure_m365/ingest/parser.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        -   name: set
+            set:
+                vadesecure.status: '{{parse_json.message.status.status}}'
+                vadesecure.substatus: '{{parse_json.message.status.substatus}}'
+    generic_email:
+        actions:
+        -   name: set
+            set:
+                email.local_id: '{{parse_json.message.id}}'
+                email.message_id: '{{parse_json.message.message_id}}'
```
Comment (PierrickV) [inline VadeSecure/vade_secure_m365/ingest/parser.yml:70]: 2 fields for the same information ?

File: VadeSecure/vade_secure_m365/_meta/fields.yml (hunk 2 above)
Comment (LeoArurault) [reply VadeSecure/vade_secure_m365/_meta/fields.yml:7]: Quotes are needed as the value contains `:` character

File: VadeSecure/vade_secure_m365/_meta/fields.yml (hunk 3 above)
Comment (LeoArurault) [reply VadeSecure/vade_secure_m365/_meta/fields.yml:27]: Quotes are needed as the value contains : character

File: VadeSecure/vade_secure_m365/ingest/parser.yml (hunk 5 above)
Comment (LeoArurault) [reply VadeSecure/vade_secure_m365/ingest/parser.yml:70]: It is not the same information :mag_right:

File: VadeSecure/vade_secure_m365/_meta/fields.yml (hunk 4 above)
Comment (LeoArurault) [reply VadeSecure/vade_secure_m365/_meta/fields.yml:32]: Done

File: VadeSecure/vade_secure_m365/tests/email.json (hunk 1 above)
Comment (LeoArurault) [reply VadeSecure/vade_secure_m365/tests/email.json:11]: `action` is used in `categorization` stage
For other, we can still add them if necessary when TDR will develop specific rules

File: VadeSecure/vade_secure_m365/ingest/parser.yml (hunk 5 above)
Comment (PierrickV) [reply VadeSecure/vade_secure_m365/ingest/parser.yml:70]: :man_facepalming:

**Review by PierrickV — APPROVED** [review]
Nice work !

**Pull request merged**